x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -155,4 +287,270 @@ export const RelevanceTuningLogic = kea<
+      if (!query) {
+        return actions.clearSearchResults();
+      }
+      actions.setResultsLoading(true);
+      const url = `/api/app_search/engines/${engineName}/search_settings_search`;
+
+      try {
+        const response = await http.post(url, {
+          query: {
+            query,
+          },
+          body: JSON.stringify({
+            boosts: isEmpty(boosts) ? undefined : boosts,
+            search_fields: isEmpty(searchFields) ? undefined : searchFields,
+          }),
+        });
+
+        actions.setSearchResults(response.results);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    onSearchSettingsSuccess: ({ searchSettings }) => {
+      actions.setSearchSettingsResponse(searchSettings);
+      actions.getSearchResults();
+      window.scrollTo(0, 0);
+    },
+    onSearchSettingsError: () => {
+      window.scrollTo(0, 0);
+    },
+    updateSearchSettings: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings`;
+
+      try {
+        const response = await http.put(url, {
+          body: JSON.stringify(removeBoostStateProps(values.searchSettings)),
+        });
+        setSuccessMessage(UPDATE_SUCCESS_MESSAGE);
+        actions.onSearchSettingsSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.onSearchSettingsError();
+      }
+    },
+    resetSearchSettings: async () => {
+      if (window.confirm(RESET_CONFIRMATION_MESSAGE)) {
+        const { http } = HttpLogic.values;
+        const { engineName } = EngineLogic.values;
+
+        const url = `/api/app_search/engines/${engineName}/search_settings/reset`;
+
+        try {
+          const response = await http.post(url);
+          setSuccessMessage(DELETE_SUCCESS_MESSAGE);
+          actions.onSearchSettingsSuccess(response);
+        } catch (e) {
+          flashAPIErrors(e);
+          actions.onSearchSettingsError();
+        }
+      }
+    },
+    toggleSearchField: ({ name, disableField }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: searchSettings.boosts,
+        search_fields: {
+          ...omit(searchFields, name),
+          ...(!disableField && { [name]: { weight: 1 } }),
+        },
+      });
+      actions.getSearchResults();
+    },
+    updateFieldWeight: ({ name, weight }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: searchSettings.boosts,
+        search_fields: {
+          ...searchFields,
+          [name]: {
+            ...searchFields[name],
+            weight: Math.round(weight * 10) / 10,
```

Comment:
Just a thought, maybe we should use something like [Number.toFixed()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) for rounding.  I see we also do this with boosts)

Reply:
738c18e

Reply:
738c18e